xray/services/xraywatch.go

```diff
@@ -0,0 +1,186 @@
+	resp, body, err := xws.client.SendDelete(xws.getXrayWatchURL()+"/"+watchName, nil, &httpClientsDetails)
+	if err != nil {
+		return err
+	}
+	if resp.StatusCode != http.StatusOK {
+		return errorutils.CheckError(errors.New("Artifactory response: " + resp.Status + "\n" + clientutils.IndentJson(body)))
+	}
+
+	log.Debug("Artifactory response:", resp.Status)
+	log.Info("Done deleting watch.")
+	return nil
+}
+
+// Create will create a new xray watch
+func (xws *XrayWatchService) Create(params utils.XrayWatchParams) error {
+	payloadBody, err := utils.CreateBody(params)
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+
+	content, err := json.Marshal(payloadBody)
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+
+	httpClientsDetails := xws.XrayDetails.CreateHttpClientDetails()
+	artUtils.SetContentType("application/json", &httpClientsDetails.Headers)
+	var url = xws.getXrayWatchURL()
+	var resp *http.Response
+	var respBody []byte
+
+	log.Info("Creating watch...")
+	resp, respBody, err = xws.client.SendPost(url, content, &httpClientsDetails)
+
+	log.Info("Finished request")
+	if err != nil {
+		log.Info("err: " + err.Error())
+		log.Error("error")
+		return err
+	}
+	log.Info("statuscode: " + resp.Status)
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusCreated {
+		return errorutils.CheckError(errors.New("Artifactory response: " + resp.Status + "\n" + clientutils.IndentJson(respBody)))
+	}
+	log.Debug("Artifactory response:", resp.Status)
+	log.Info("Done creating watch.")
+	return nil
+}
+
+// Update will update an existing Xray watch by name
+// It will error if no watch can be found by that name.
+func (xws *XrayWatchService) Update(params utils.XrayWatchParams) error {
+	payloadBody, err := utils.CreateBody(params)
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+
+	// the update payload must not have a name
+	payloadBody.GeneralData.Name = ""
+
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+
+	content, err := json.Marshal(payloadBody)
+	if err != nil {
+		return errorutils.CheckError(err)
+	}
+
+	httpClientsDetails := xws.XrayDetails.CreateHttpClientDetails()
+	artUtils.SetContentType("application/json", &httpClientsDetails.Headers)
+	var url = xws.getXrayWatchURL() + "/" + params.Name
+	var resp *http.Response
+	var respBody []byte
+
+	log.Info("Updating watch...")
+	resp, respBody, err = xws.client.SendPut(url, content, &httpClientsDetails)
+
+	log.Info("Finished request")
+	if err != nil {
+		log.Info("err: " + err.Error())
+		log.Error("error")
+		return err
+	}
+	log.Info("statuscode: " + resp.Status)
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusCreated {
+		return errorutils.CheckError(errors.New("Artifactory response: " + resp.Status + "\n" + clientutils.IndentJson(respBody)))
+	}
+	log.Debug("Artifactory response:", resp.Status)
+	log.Info("Done updating watch.")
+	return nil
+}
+
+// Get retrieves the details about an Xray watch by name
```

Comment:
// Get retrieves the details about an Xray watch by name --> // Get retrieves the details of an Xray watch by its name.